src/components/CartButton.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { AppState } from '../store';
+import { Product } from '../types/product';
+
+const CartButton = () => {
+ 
+  const products = useSelector<AppState, Product[]>(state => state.cartState.products);
```

Comment:
This is definitely valid, but one thing that can be a pain as it grows is having to supply the second generic (shape of the return from the selector). Instead, we often rely on inference:

```ts
  const products = useSelector((state: AppState) => state.cartState.products);
```

Reply:
Thanks Tim, I'll fix this up.